gc/base/Configuration.hpp

```diff
@@ -93,6 +93,23 @@ class MM_Configuration : public MM_BaseVirtual
 	bool initializeHeapRegionDescriptor(MM_EnvironmentBase *env, MM_HeapRegionDescriptor *region) { return _delegate.initializeHeapRegionDescriptorExtension(env, region); }
 	void teardownHeapRegionDescriptor(MM_EnvironmentBase *env, MM_HeapRegionDescriptor *region) { _delegate.teardownHeapRegionDescriptorExtension(env, region); }
 
+	MMINLINE MM_GCWriteBarrierType getBarrierType() { return _writeBarrierType; }
+
+	MMINLINE bool
+	isSnapshotAtTheBeginningBarrierEnabled()
+	{
+		return getBarrierType() == gc_modron_wrtbar_satb_and_oldcheck ||
+				getBarrierType() == gc_modron_wrtbar_satb;
+	}
+
+	MMINLINE bool
+	isIncrementalUpdateBarrierEnabled()
+	{
+		return getBarrierType() == gc_modron_wrtbar_cardmark_and_oldcheck ||
+				getBarrierType() == gc_modron_wrtbar_oldcheck ||
```

Comment:
it would be only the 2 conditions with _cardmark_. Remove the pure gc_modron_wrtbar_oldcheck case.